### DISCUSSION TITLE
dbus-next unmaintained, switch to dbus-fast?

### DISCUSSION THREAD

**Discussion by raj-magesh** [qtile/qtile#5112] [Ideas]
In [this PR](https://github.com/NixOS/nixpkgs/pull/360361), the NixOS maintainer of `dbus-next` points out that it's unmaintained and that the tests are flaky. They recommend switching to `dbus-fast`: https://github.com/Bluetooth-Devices/dbus-fast

**Comment by elParaguayo** [comment]
Thanks for posting this. I've been aware of the fact that `dbus-next` is unmaintained for quite some time. Similarly, I'm aware that `dbus-fast` seems to be the de facto replacement for it.

My sense is we should move to dbus-fast but I'm not sure the best way to do it. We could either rip out dbus-next entirely and replace it with dbus-fast and get loads of issues posted about things breaking because users typically don't read changelogs. Or, we retain dbus-next support but we add deprecation warnings too. That may work if the API is the same but my understanding is that there are some slight differences.

@tych0 any preferences? I'm happy to implement the changes, whichever route we go down, given that I added dbus-next support originally.

**Reply by tych0** [reply]
I think it's fine to just wholesale switch. Our widgets give pretty good error messages if imports are missing, hopefully that will be enough for people to figure it out.

That said, if you want to try and support both, I'm fine with that too. But it does seem like more of a hassle than answering a handful of github issues.

**Reply by elParaguayo** [reply]
Thanks. Leave this one with me to implement.

**Comment by elParaguayo** [comment]
Closing this as we've now migrated to dbus-fast.
